lib/kubernetes-deploy/kubernetes_resource/stateful_set.rb

```diff
@@ -0,0 +1,61 @@
+# frozen_string_literal: true
+require 'kubernetes-deploy/kubernetes_resource/pod_set_base'
+module KubernetesDeploy
+  class StatefulSet < PodSetBase
+    TIMEOUT = 5.minutes
+    ONDELETE = 'OnDelete'
+    attr_reader :pods
+
+    def sync
+      raw_json, _err, st = kubectl.run("get", type, @name, "--output=json")
+      @found = st.success?
+
+      if @found
+        stateful_data = JSON.parse(raw_json)
+        @status_data = stateful_data["status"]
+        @rollout_data = stateful_data["status"].slice("replicas")
+        @update_strategy = stateful_data["spec"]["updateStrategy"]["type"]
+        @status = @rollout_data.map { |state_replicas, num| "#{num} #{state_replicas.chop.pluralize(num)}" }.join(", ")
+        @pods = find_pods(stateful_data)
+      else # reset
+        @rollout_data = { "replicas" => 0 }
```

Comment:
This should reset `@status_data` instead in a way that will not pass `deploy_succeeded?`